libnmstate/state.py

```diff
@@ -32,6 +32,68 @@
 from libnmstate.schema import Interface
 from libnmstate.schema import InterfaceState
 from libnmstate.schema import InterfaceType
+from libnmstate.schema import Route
+
+
+class RouteEntry(object):
+
+    # Mapping dict key name to RouteEntry property name
+    KEY_PROPETY_MAP = {
+        Route.STATE: Route.STATE.replace('-', '_'),
+        Route.TABLE_ID: Route.TABLE_ID.replace('-', '_'),
+        Route.METRIC: Route.METRIC.replace('-', '_'),
+        Route.NEXT_HOP_ADDRESS: Route.NEXT_HOP_ADDRESS.replace('-', '_'),
+        Route.DESTINATION: Route.DESTINATION.replace('-', '_'),
+        Route.NEXT_HOP_INTERFACE: Route.NEXT_HOP_INTERFACE.replace('-', '_'),
+    }
+
+    def __init__(self, route):
+        self.table_id = route.get(Route.TABLE_ID)
```

Comment:
you could use the `KEY_PROPERTY_MAP` with `setattr()` here to avoid having the list of properties twice.

Reply:
The pylint or flake8 will contain about later I check `self.table_id` with defined in `__init__`.